src/viewport.cpp

```diff
@@ -831,15 +829,20 @@ void AddChildSpriteScreen(SpriteID image, PaletteID pal, int x, int y, bool tran
 	assert((image & SPRITE_MASK) < MAX_SPRITES);
 
 	/* If the ParentSprite was clipped by the viewport bounds, do not draw the ChildSprites either */
-	if (_vd.last_child == nullptr) return;
+	if (_vd.last_child == LAST_CHILD_NONE) return;
 
 	/* make the sprites transparent with the right palette */
 	if (transparent) {
 		SetBit(image, PALETTE_MODIFIER_TRANSPARENT);
 		pal = PALETTE_TO_TRANSPARENT;
 	}
 
-	*_vd.last_child = (uint)_vd.child_screen_sprites_to_draw.size();
+	int32_t child_id = static_cast<int32_t>(_vd.child_screen_sprites_to_draw.size());
+	if (_vd.last_child >= 0) {
```

Comment:
I'm a bit conflicted about this line. On its own it looks very reasonable to guard indexing into an array with a non-negative value. However, `LAST_CHILD_PARENT` is only set and never read, and here the "meaning" seems to be `if (_vd.last_child != LAST_CHILD_PARENT)`. That'd make me think... how is `LAST_CHILD_NONE` then handled as that's a negative value, just to discover that if that's the case you leave the function a couple of lines earlier.

So yeah, not sure. Either question where `LAST_CHILD_PARENT` is read or question why `LAST_CHILD_NONE` is not checked for in this `if`.

Reply:
Decided to go with `!= LAST_CHILD_PARENT` as that is the actual intention and it took me a moment to register that already.

If it was for bounds checking then `>= 0` is not a sufficient check.